code/game/objects/structures/signs.dm

```diff
@@ -23,6 +28,48 @@
 		if(WEST)
 			pixel_x = -30
 
+//automatically adjust place and offset to make sure sign isn't floating in the middle of nowhere
+/obj/structure/sign/proc/place_poster()
+	var/turf/closed/applied_turf = get_step(loc, dir)
+	if(istype(applied_turf))
+		switch(dir)
+			if(NORTH)
+				pixel_y = 32
+			if(SOUTH)
+				pixel_y = -32
+			if(EAST)
+				pixel_x = 30
+			if(WEST)
+				pixel_x = -30
+		return
+	var/turf/locatedturf = get_turf(loc)
+	if(isclosedturf(locatedturf))
+		return
+	for(var/i in CARDINAL_ALL_DIRS)
+		locatedturf = get_step(loc, i)
+		if(isclosedturf(locatedturf))
+			switch(i)
```

Comment:
we presumably should be returning after this switch statement, otherwise we keep cycling through directions even after we found a solid wall.

Reply:
This is changed, not sure why it's not showing as outdated